include/platform.hpp

```diff
@@ -40,15 +49,17 @@
 # include <unistd.h>
 #endif
 
-// MSVC doesn't support `[static N]` for array arguments from C99 or C11
-#ifdef _MSC_VER
-# define MIN_NB_ELMS(N)
-# define ARR_QUALS(...)
-# define NONNULL(ptr) *ptr
+// C++ doesn't support `[static N]` for array arguments from C99 or C11
+#define MIN_NB_ELMS(N) // static (N)
+#define ARR_QUALS(...) // __VA_ARGS__
+#define NONNULL(ptr) *ptr // ptr[static 1]
```

Comment:
Shouldn't it be possible to use an attribute for this? (We can tackle that in a later PR, though.)

Reply:
`__attribute__((nonnull))` exists, but it's not a [standard attribute](https://en.cppreference.com/w/cpp/language/attributes), probably because C++ already has references for that. `ARR_QUALS(restrict)` likewise just has a nonstandard alternative, `__restrict__`.

Reply:
Oh, I thought it was. That said, note that `[[gnu::nonnull]]` is equivalent standard syntax, and is required to be ignored by implementations not understanding it.